### ISSUE TITLE
False positive warn when define `as_*` method with self-parameter wrapped by pointers

### ISSUE THREAD

**Issue by p0lunin** [intellij-rust/intellij-rust#8352]
## Environment

* **IntelliJ Rust plugin version:** 0.4.148.3911-203
* **Rust toolchain version:** 1.55.0 (c8dfcfe04 2021-09-06) x86_64-pc-windows-msvc
* **IDE name and version:** PyCharm 2020.3.3 Professional Edition (PY-203.7148.72)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Plugin shows warn `methods called as_* usually take self by reference or self by mutable reference; consider choosing a less ambiguous name`, but method take `self` by reference.

![image](https://user-images.githubusercontent.com/48922415/147855588-61ec147e-f47b-4bdb-9fc7-c769ea21d6f0.png)

## Steps to reproduce
```rust
struct Foo;
impl Foo {
    pub fn as_something(self: &Rc<Self>) -> i32 {
        0
    }
}
```

**Comment by SilvaQ** [comment]
what is this about from_*:

```rust
pub trait Alphabet {
   fn from_ug(&self, word: &str) -> String;
}
```
<img width="709" alt="image" src="https://user-images.githubusercontent.com/46706370/153695699-d5eeae37-25d8-4460-ab5c-406f2fcc276e.png">
